### PR TITLE
khal: fix contact integration

### PR DESCRIPTION
**Pull request by felipeqq2**
### Description

Currently, birthday calendars added to Khal via `accounts.contact` throw `error: attribute 'type' missing`. This PR:

- Fixes a bug to correctly add `khal.type = "birthdays"` to these calendars;
- Moves options `color`/`priority` to global Khal configuration (these options can also be set for birthday calendars, and Nix currently throws an error if non-defined);
- Adds multiple calendars for each contact collection (fixing an issue similar to #4531):
  - New option `khal.collections`;
  - Also supports null-collections (as seen [here](https://github.com/nix-community/home-manager/issues/4531#issuecomment-1748332523));

### Checklist

- [X] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@teto (git blame)

### CODE REVIEW

File: modules/programs/khal-accounts.nix [hunk 1]
```diff
@@ -13,5 +13,39 @@ with lib;
...
+        Color in which events in this calendar are displayed.
+      '';
+      example = "light green";
+    };
+
+    priority = mkOption {
+      type = types.int;
+      default = 10;
+      description = ''
+        Priority of a calendar used for coloring.
```
Comment (teto) [inline modules/programs/khal-accounts.nix:47]: could you mention which is highest priority, eg.
`(priority 0 is shown before priority 10)`

**Comment by teto** [comment]
could you rebase to trigger CI please ? I dont use accounts.contact but I am willing to try the PR this week.

File: modules/programs/khal-accounts.nix (hunk 1 above)
Comment (felipeqq2) [reply modules/programs/khal-accounts.nix]: Fixed in f6c4426.

**Comment by felipeqq2** [comment]
I don't think the failing tests were caused by my changes (same issue with #5005)

**Review by teto — APPROVED** [review]
can you confirm this is backwards compatible ?

**Comment by felipeqq2** [comment]
> can you confirm this is backwards compatible ?

Yes (AFAICT). `priority` is now an existing option with a default value (`10`) for contact calendars, but that's what was being used by khal anyway.

**Pull request merged**